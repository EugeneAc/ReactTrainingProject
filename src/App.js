import React from "react";
import ReactDOM from "react-dom";

const App = () => {
  return (
    <div>
      <p>React here!</p>
    </div>
  );
};

function App1() {
  return (
    <div className="App">
      <h1>Hello CodeSandbox</h1>
    </div>
  );
}

class Hello extends React.Component {
  render() {
    return (
      <div className="App">
        <h1>Hello CodeSandbox2</h1>
      </div>
    );
  }
}

const element = React.createElement("h1", {}, "Hello CodeSandbox3");

class Message extends React.PureComponent {
  render() {
    return (
      <div className="App">
        <h1>Hello CodeSandbox4</h1>
      </div>
    );
  }
}

export default App;

ReactDOM.render(<App />, document.getElementById("app"));

ReactDOM.render(<App1 />, document.getElementById("element1"));

ReactDOM.render(<Hello />, document.getElementById("element2"));

ReactDOM.render(element, document.getElementById("element3"));

ReactDOM.render(<Message />, document.getElementById("element4"));