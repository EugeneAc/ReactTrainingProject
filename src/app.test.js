import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import SearchResults from './components/searchResultString.jsx';
import SearchContainer from './components/searchContainer';
import App from './App';
import TogleBtns from './components/toggleBtns.jsx';

configure({ adapter: new Adapter() });

describe('Snapshot', () => {
it('App matches the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

  describe('toggleBtns', () => {
    it('Should switch state.className to left', () => {
      const wrapper = shallow(<TogleBtns changeActive={() => {}}/>);
      const btnLeft = wrapper.find('button').at(0);
      btnLeft.simulate('click');
      expect(wrapper.state().className).toEqual('left');
    });

    it('Should switch state.className to right', () => {
      const wrapper = shallow(<TogleBtns changeActive={() => {}}/>);
      const btnRight = wrapper.find('button').at(1);
      btnRight.simulate('click');
      expect(wrapper.state().className).toEqual('right');
    });
  });

  describe('App', () => {
    it('Should change state.selectedMovie to 2', () => {
      const wrapper = shallow(<App />);
      wrapper.instance().onChangeSelected('2')
      expect(wrapper.state().selectedMovie).toEqual('2');
    });

    it('Should change state.searchParam Title and Genre', () => {
      const wrapper = shallow(<App />);
      wrapper.instance().onChangeSearchParam('left')
      expect(wrapper.state().searchParam).toEqual('Title');
      wrapper.instance().onChangeSearchParam('new')
      expect(wrapper.state().searchParam).toEqual('Genre');
    });

  });

  describe('SearchContainer', () => {
    it('Should change state.searchValue to newValue', () => {
      const wrapper = shallow(<SearchContainer onChangeSearchParam={() => {}} onSearch={() => {}}/>);
      var e = {
        target:{
          value: "newValue"
      }}
      wrapper.instance().onTextChanged(e)
      expect(wrapper.state().searchValue).toEqual('newValue');
    });
  });

  