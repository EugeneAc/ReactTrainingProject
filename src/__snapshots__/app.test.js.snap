// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot App matches the snapshot 1`] = `
<div
  className="new-class"
>
  <h2>
    Find your movie
  </h2>
  <input
    className="search-field"
    onChange={[Function]}
    placeholder="Search"
  />
  <button
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-search"
    />
  </button>
  <div
    className="under-search"
  >
    <div
      className="inline"
    >
      <span>
        Search by
      </span>
      <div
        className="toggle-btms inline"
      >
        <button
          className="left"
          onClick={[Function]}
        >
          Title
        </button>
        <button
          className="left"
          onClick={[Function]}
        >
          Genre
        </button>
      </div>
    </div>
  </div>
  <div
    className="search-result"
  >
    <div
      className="inline"
    >
      <span>
        3
         Movies found
      </span>
    </div>
  </div>
  <div
    className=""
  >
    <div
      className="alignRight sorting-panel"
    >
      Sort by
      <div
        className="toggle-btms inline"
      >
        <button
          className="left"
          onClick={[Function]}
        >
          Release date
        </button>
        <button
          className="left"
          onClick={[Function]}
        >
          Rating
        </button>
      </div>
    </div>
    <div
      className="movies-container"
    >
      <div
        className="movie-face-plate"
        onClick={[Function]}
      >
        <img
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSJjJ0oj9DdOyVjYyDREytOES2pKEDgUtXDvNeaDnflAoXfg6I4"
        />
        <div
          className="movie-name inline"
        >
          Kill Bill
        </div>
        <div
          className="movie-year alignRight"
        >
          2003
        </div>
        <div
          className="movie-genre"
        >
          Action
        </div>
      </div>
      <div
        className="movie-face-plate"
        onClick={[Function]}
      >
        <img
          src="https://upload.wikimedia.org/wikipedia/en/thumb/c/c4/Kill_Bill_Volume_2.png/220px-Kill_Bill_Volume_2.png"
        />
        <div
          className="movie-name inline"
        >
          Kill Bill 2
        </div>
        <div
          className="movie-year alignRight"
        >
          2004
        </div>
        <div
          className="movie-genre"
        >
          Action
        </div>
      </div>
      <div
        className="movie-face-plate"
        onClick={[Function]}
      >
        <img
          src="https://i.pinimg.com/originals/bd/71/8c/bd718cb3d0932ff76eafc649921772a9.jpg"
        />
        <div
          className="movie-name inline"
        >
          Kill Bill 3
        </div>
        <div
          className="movie-year alignRight"
        >
          2018
        </div>
        <div
          className="movie-genre"
        >
          Action
        </div>
      </div>
    </div>
  </div>
  <div
    className="movie-info"
  >
    <img
      src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSJjJ0oj9DdOyVjYyDREytOES2pKEDgUtXDvNeaDnflAoXfg6I4"
    />
    <div
      className="movie-desc"
    >
      <div
        className="movie-name-info inline"
      >
        Kill Bill
      </div>
      <div
        className="movie-length-info inline"
      />
      <div
        className="movie-year-info inline"
      >
        2003
      </div>
      <div
        className="movie-decs-info inline"
      >
        Kill Bill: Volume 1 is a 2003 American martial arts film written and directed by Quentin Tarantino. It stars Uma Thurman as the Bride, who swears revenge on a team of assassins (Lucy Liu, Michael Madsen, Daryl Hannah, and Vivica A. Fox) and their leader Bill (David Carradine) after they try to kill her and her unborn child. Her journey takes her to Japan, where she battles the Tokyo yakuza.
      </div>
    </div>
  </div>
</div>
`;
